### ISSUE TITLE
Strict mode?

### ISSUE THREAD

**Issue by stubar** [loeffel-io/ls-lint#32]
At the moment the rules kick in if an extension has been matched. It would be nice to stipulate that only files of a certain extension are allowed (with case rules enforced).

**Comment by loeffel-io** [comment]
Please provide more infos

**Comment by stubar** [comment]
At the moment the rules kick in if an extension has been matched.  It would be nice to stipulate that only files of a certain extension are allowed (with case rules enforced).

**Comment by stubar** [comment]
Sorry I seem to have pressed the return key prematurely!

**Comment by loeffel-io** [comment]
Could you please make an example about that

**Comment by stubar** [comment]
I have this setup for my components folder...

```ls:
  src/components/**:
  .dir: PascalCase
  .ts: camelCase
  .test.ts: camelCase
  .tsx: PascalCase
  .test.tsx: PascalCase
  .stories.test.tsx: PascalCase
  .d.ts: kebab-case```

**Comment by loeffel-io** [comment]
Got it! Like https://github.com/loeffel-io/ls-lint/issues/30 we could create something like `not_exists` rule 

What do you think?

**Comment by stubar** [comment]
Yes that sounds like it would work, you could have exists(count).  Same rule could enforce no occurrences `exists(0)` and may theres a case where you only ever want one occurrence of that extension `exists(1)`.

Also I thought as a work around for my js rule, I could for now set a regex that insists on a char not allowed by filesystems.

**Comment by loeffel-io** [comment]
yea, i will think about that some time 

I think this will be added in v2 👍

**Comment by Igorkowalski94** [comment]
As an alternative, you can use [eslint-plugin-project-structure](https://github.com/Igorkowalski94/eslint-plugin-project-structure), you will have eslint errors in real time.

features:
✅ Validation of project structure (Any files/folders outside the structure will be considered an error).
✅ Validation of folder and file names.
✅ Name case validation.
✅ Name regex validation.
✅ File extension validation (Support for all extensions).
✅ Inheriting the parent's name (The child inherits the name of the folder in which it is located).
✅ Folder recursion (You can nest a given folder structure recursively).
✅ Forcing a nested/flat structure for a given folder.

**Comment by ccoVeille** [comment]
For anyone interested. You can now use `exists:0` as a rule. Please check the documentation

**Comment by loeffel-io** [comment]
@ccoVeille `v2.3.0` is not released yet - there is one single piece that is missing

**Comment by ccoVeille** [comment]
Author opened the issue 4 years ago. I assumed he could wait a few more days/week.

But you are right, it's not yet available or documented

**Comment by loeffel-io** [comment]
As far i can tell this is one of the most challenging tasks for me - its beautiful how wildcard extensions, selected files and exists working together magically

**Comment by loeffel-io** [comment]
This feature is now available in the `v2.3.0-beta.1` release. 
Please test it and let me know if there are any issues ❤️ 

The upcoming v2.3.0 blog post: https://ls-lint.org/blog/announcements/v2.3.0.html
The docs: https://ls-lint.org/2.3/getting-started/introduction.html
The release notes: https://github.com/loeffel-io/ls-lint/releases/tag/v2.3.0-beta.1
